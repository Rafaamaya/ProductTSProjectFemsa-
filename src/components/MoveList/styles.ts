import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    backgroundColor: 'white',
    flexDirection: 'row',
    marginBottom: 5,
  },
  image: {
    width: 45,
    height: 45,
    borderRadius: 10,
  },
  containerProduct: {
    marginLeft: 11,
  },
  titleProduct: {
    fontWeight: '800',
    fontSize: 14,
    color:'#000000',
    fontFamily: 'avenir',
  },
  textDate: {
    fontWeight: '400',
    fontSize: 12,
    fontFamily: 'avenir',
  },
  containerPoints: {
    justifyContent: 'flex-end',
    flex: 1,
    alignItems: 'center',
    flexDirection: 'row',
  },
  textPoints: {
    fontWeight: '800',
    fontSize: 16,
    fontFamily: 'avenir',
    color:'#000000',
  },
  textIcon: {
    marginRight: 5,
    marginLeft: 16,
    fontWeight: '400',
    fontSize: 16,
    color:'#000000',
  },
});
