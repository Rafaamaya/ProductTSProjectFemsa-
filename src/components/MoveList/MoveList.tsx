/* eslint-disable react-native/no-inline-styles */
import React from 'react';
import {Image, Text, TouchableOpacity, View} from 'react-native';
import styles from './styles';
import {Product} from '../../interfaces/productInterface';
import { formatFeedDate } from '../../utils/timeUtils';
import { useNavigation } from '@react-navigation/native';
import { StackNavigationProp } from '@react-navigation/stack';
import { RootStackParams } from '../../navigator/StackNavigator';



type DetailscreenNavigationProp = StackNavigationProp<RootStackParams, 'DetailsScreen'>;

interface Props {
  item: Product;
}

const MoveList = ({item}: Props) => {

  const navigation = useNavigation<DetailscreenNavigationProp>();

  const handleOnPres = () =>{
    navigation.navigate('DetailsScreen' , { item });

  };
  return (
    <TouchableOpacity style={styles.container} onPress={handleOnPres}>
      <View>
        <Image style={styles.image} source={{uri: item.image}} />
      </View>
      <View style={styles.containerProduct}>
        <Text style={styles.titleProduct}>{item.product}</Text>
        <Text style={styles.textDate}>{formatFeedDate(item.createdAt)}</Text>
      </View>
      <View style={styles.containerPoints}>
        <Text
          style={[
            styles.textPoints,
            {color: item.is_redemption ? '#00B833' : '#FF0000'},
          ]}>
          {item.is_redemption ? '+ ' : '- '}
        </Text>
        <Text style={styles.textPoints}>{item.points}</Text>
        <Text style={styles.textIcon}> {'>'} </Text>
      </View>
    </TouchableOpacity>
  );
};

export default MoveList;
