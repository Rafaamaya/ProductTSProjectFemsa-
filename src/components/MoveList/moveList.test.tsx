import React from 'react';
import renderer from 'react-test-renderer';
import MoveList from './MoveList';

export const MOCKED_DATA = {
  createdAt: '2022-12-09T10:20:00.909Z',
  product: 'Fantastic Granite Bacon',
  points: 100,
  image: 'https://loremflickr.com/640/480/technics',
  isRedemption: false,
  id: '3',
};

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: jest.fn(),
    }),
  };
});

describe('Pruebas en <MoveList/>', () => {
  test('debe de hacer match con el snapchot', () => {
    const tree = renderer
      .create(<MoveList item={MOCKED_DATA as any} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

