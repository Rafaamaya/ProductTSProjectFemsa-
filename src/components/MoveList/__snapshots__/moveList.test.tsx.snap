// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MoveList/> debe de hacer match con el snapchot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "flexDirection": "row",
      "marginBottom": 5,
      "opacity": 1,
    }
  }
>
  <View>
    <Image
      source={
        Object {
          "uri": "https://loremflickr.com/640/480/technics",
        }
      }
      style={
        Object {
          "borderRadius": 10,
          "height": 45,
          "width": 45,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "marginLeft": 11,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "avenir",
          "fontSize": 14,
          "fontWeight": "800",
        }
      }
    >
      Fantastic Granite Bacon
    </Text>
    <Text
      style={
        Object {
          "fontFamily": "avenir",
          "fontSize": 12,
          "fontWeight": "400",
        }
      }
    >
      9 de Diciembre, 2022
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-end",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "avenir",
            "fontSize": 16,
            "fontWeight": "800",
          },
          Object {
            "color": "#FF0000",
          },
        ]
      }
    >
      - 
    </Text>
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "avenir",
          "fontSize": 16,
          "fontWeight": "800",
        }
      }
    >
      100
    </Text>
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 16,
          "fontWeight": "400",
          "marginLeft": 16,
          "marginRight": 5,
        }
      }
    >
       
      &gt;
       
    </Text>
  </View>
</View>
`;
